src/java.base/unix/native/libnio/ch/Net.c

```diff
@@ -295,6 +299,16 @@ Java_sun_nio_ch_Net_socket0(JNIEnv *env, jclass cl, jboolean preferIPv6,
             close(fd);
             return -1;
         }
+        if (level == IPPROTO_IPV6) {
+            if ((setsockopt(fd, level, IPV6_MULTICAST_ALL, (char*)&arg, sizeof(arg)) < 0) &&
+                (errno != ENOPROTOOPT)) {
+                JNU_ThrowByNameWithLastError(env,
+                                         JNU_JAVANETPKG "SocketException",
+                                         "Unable to set IPV6_MULTICAST_ALL");
+                close(fd);
+                return -1;
+            }
```

Comment:
It might be better to move this to just below the attempt to set IPV6_MULTICAST_HOPS as that code only runs for IPv6 datagrams.

That said, I'm a bit concerned that this will fail on older kernels. Are you sure the error is ENOPROTOOPT?

Reply:
Yes, the error is the same as for the IPv4 option. If you like I can change test to call the option and check an exception is not thrown in those cases.